pinot-connectors/pinot-connector-kafka-base/src/test/java/org/apache/pinot/core/realtime/impl/kafka/KafkaJSONMessageDecoderTest.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.realtime.impl.kafka;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.util.HashMap;
+import org.apache.pinot.common.data.FieldSpec;
+import org.apache.pinot.common.data.Schema;
+import org.apache.pinot.common.data.TimeFieldSpec;
+import org.apache.pinot.core.data.GenericRow;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class KafkaJSONMessageDecoderTest {
+
+  private static ObjectMapper objectMapper = new ObjectMapper();
+
+  @Test
+  public void testJsonDecoderWithoutOutgoingTimeSpec()
+      throws Exception {
+    Schema schema = Schema.fromFile(new File(
+        getClass().getClassLoader().getResource("data/test_sample_data_schema_without_outgoing_time_spec.json")
+            .getFile()));
+    BufferedReader reader = new BufferedReader(
```

Comment:
Can we make this try-with-resource and remove the `reader.close()` in `testJsonDecoder()`? Same for other tests.

Reply:
do you mean something like: 
```
    try (BufferedReader reader = new BufferedReader(
        new FileReader(getClass().getClassLoader().getResource("data/test_sample_data.json").getFile()))) {
      testJsonDecoder(schema, reader);
    }